lib/utility.js

```diff
@@ -74,43 +205,15 @@ function getAllActiveProducts(offset, pageSize) {
  */
 function getProductsByType(typeId, offset, pageSize) {
   const types = _getProductTypeMapping();
-
-  return new Promise((resolve, reject) => {
-    // query params
-    const data = {
+  let data = {
       "Action": "GetProductsByType",
       "TypeId": typeId
-    };
-    if (offset) {
-      data["StartId"] = offset;
-    }
-    if (pageSize) {
-      data["pageSize"] = pageSize;
-    }
-
-    // query server
-    wx.request({
-      url: _marketUrl,
-      data,
-      method: "POST",
-      header: {
-        "content-type": "application/x-www-form-urlencoded",
-      },
-      dataType: "json",
-      success: (res) => {
-        // verify everything went well on the server side
-        res = res.data;
-        if (res.ErrorCode != 1) {
-          fail({
-            ErrorCode: res.ErrorCode,
-            ErrorMessage: res.ErrorMessage
-          });
-        } else {
-          resolve(res.Products.map((x) => new Product(x)));
-        }
-      },
-      fail: (res) => reject("WX request failed")
-    });
+  };
+  if (offset) data["StartId"] = offset;
+  if (pageSize) data["pageSize"] = pageSize;
+  return requestPromise(marketUrl, data)
+  .then(function(res){

```

Comment:
Same "then()“ problem.
Should return a new promise resolved when the promise returned by reqestPromise is resolved.
We should also handle the failure case for this new promise.